src/common/libkvs/kvs_commit.c

```diff
@@ -61,12 +61,23 @@ flux_future_t *flux_kvs_fence (flux_t *h, const char *ns, int flags,
             return NULL;
     }
 
+    if (flags & FLUX_KVS_SYNC
+        && strcmp (ns, KVS_PRIMARY_NAMESPACE) != 0) {
+        errno = EINVAL;
+        return NULL;
+    }
+
```

Comment:
This check should be in the kvs module, not in the client, in case the RPCs are made directly, for example by a language binding that cannot readily wrap Flux futures.